tee/src/lib.rs

```diff
@@ -0,0 +1,594 @@
+		AddedEnclave {
+			account: T::AccountId,
+			api_uri: Vec<u8>,
+			enclave_id: EnclaveId,
+		},
+		/// An enclave got unregistered
+		UnregisterEnclave {
+			account_id: T::AccountId,
+			enclave_id: EnclaveId,
+		},
+		/// An enclave moved for unregistration to a queue
+		MovedForUnregistration {
+			account_id: T::AccountId,
+			enclave_id: EnclaveId,
+		},
+		/// An enclave got assigned to a cluster
+		AssignedEnclave {
+			enclave_id: EnclaveId,
+			cluster_id: ClusterId,
+		},
+		/// An enclave got assigned
+		UnAssignedEnclave {
+			enclave_id: EnclaveId,
+		},
+		/// An enclave got updated
+		UpdatedEnclave {
+			enclave_id: EnclaveId,
+			api_uri: Vec<u8>,
+		},
+		/// Enclave force updated
+		EnclaveUpdated { enclave_id: EnclaveId, enclave_address: Vec<u8>, api_uri: Vec<u8> },
+		/// New cluster got added
+		AddedCluster {
+			cluster_id: ClusterId,
+		},
+		/// Cluster got removed
+		RemovedCluster {
+			cluster_id: ClusterId,
+		},
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Unknown enclaveId
+		UnknownEnclaveId,
+		/// Unknown ClusterId
+		UnknownClusterId,
+		/// Account does not associated with an enclave
+		NotEnclaveOwner,
+		/// Enclave address registered to an account
+		EnclaveAddressAlreadyRegisteredtoTheAccount,
+		/// Enclave URI is short
+		UriTooShort,
+		/// Enclave URI is long
+		UriTooLong,
+		/// Maximum enclaves reached
+		EnclaveIdOverflow,
+		/// Maximum clusters reached
+		ClusterIdOverflow,
+		/// Cluster is already full, cannot assign any enclaves
+		ClusterIsAlreadyFull,
+		/// Enclave already assigned to a cluster
+		EnclaveAlreadyAssigned,
+		/// Enclave not assigned to a cluster
+		EnclaveNotAssigned,
+		/// Enclave does not exists
+		EnclaveDoesNotExists,
+		/// Cannot assign to same cluster
+		CannotAssignToSameCluster,
+		/// Internal logical error
+		InternalLogicalError,
+		/// Assigning an operator to an invalid enclaveId
+		AssigningOperatorForUnknownEnclaveId,
+		/// Unknown enclave operator account
+		UnknownEnclaveOperatorAccount,
+		/// Invalid IAS sign certificate
+		InvalidIASSigningCert,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+
+		#[pallet::weight(T::WeightInfo::register_enclave())]
+		pub fn register_enclave(
+			origin: OriginFor<T>,
+			enclave_address: Vec<u8>,
+			api_uri: Vec<u8>,
+		) -> DispatchResultWithPostInfo {
+			let account = ensure_signed(origin)?;
+
+			ensure!(api_uri.len() < T::MaxUriLen::get().into(), Error::<T>::UriTooLong);
+			ensure!(api_uri.len() > T::MinUriLen::get().into(), Error::<T>::UriTooShort);
+
+			let has_registered_index = EnclaveIndex::<T>::get(account.clone());
+
+			// Verify if the origin does not already have an enclave address associated with it.
+			match has_registered_index {
+				Some(enc_id) => {
+					let enc = EnclaveRegistry::<T>::get(enc_id).unwrap();
+					ensure!(
+						enc.enclave_address != enclave_address,
+						Error::<T>::EnclaveAddressAlreadyRegisteredtoTheAccount
+					)
+				}
+				_ => {}
+			}
```

Comment:
Refactor using ensure and EnclaveIndex::<T>::get(account.clone()).is_none()

Reply:
Here the code does not need to throw an error if its None, So can't use ensure. The check needs to happen is to see the new enclave address has been registered previously